exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -397,17 +412,42 @@ private static Exemplar toProtoExemplar(
         Exemplar.newBuilder()
             .setValue(exemplar.getValue())
             .setTimestamp(convertTimestamp(exemplar.getTimestamp()));
+    @javax.annotation.Nullable String traceId = null;
+    @javax.annotation.Nullable String spanId = null;
     for (Map.Entry<String, String> attachment : exemplar.getAttachments().entrySet()) {
-      // TODO: add support for special attachments once Stackdriver Monitoring accepts them.
+      String key = attachment.getKey();
+      String value = attachment.getValue();
+      if (ExemplarUtils.ATTACHMENT_KEY_TRACE_ID.equals(key)) {
```

Comment:
I know that currently this package is not used, and I am curious to know if it is better to extend the Exemplar to support SpanContext directly to avoid these 2 keys, what do you think?

Something like https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/stats/MeasureMap.java#L67
```java
public MeasureMap putAttachment(SpanContext spanContext);
```

When we export in metrics we can export using these headers probably, what do you think? I think I am mostly worried about recording path, but in our Metrics if we can keep it as <string, string>.

Reply:
I remember we discussed this approach before. That would introduce dependency on Tracing APIs to Stats APIs, which is not ideal. That's also the main reason that we chose <String, String> as exemplar attachments.

An alternative is to add the API `putAttachment(SpanContext spanContext)` in a contrib util package.

Reply:
Other option is to have:

abstract class LazyAttachmentValue {
  public String getValue();
}

putAttachment(String key,  LazyAttachmentValue value)

And then in the util we create the LazyAttachmentValue for SpanContext, also we make the SpanContext attachment to be one line, which I think we should do even in Logs if you ask me. Maybe that sail is shipped already.

Reply:
I file an issue https://github.com/census-instrumentation/opencensus-java/issues/1775 to keep track of this, since it involves new API support. This PR is mainly on exporter changes.

Reply:
Yeah, but before we add this support we can guarantee that nobody uses the keys defined in the exemplar util so I think we should do that first, this way we can simply do a breaking change in the exemplar util.

Reply:
OK that makes sense to me, putting this PR on hold for now.